src/main/java/nextstep/subway/domain/Line.java

```diff
@@ -0,0 +1,88 @@
+package nextstep.subway.domain;
+
+import nextstep.subway.dto.LineRequest;
+
+import javax.persistence.*;
+import java.util.Objects;
+
+@Entity
+public class Line extends BaseEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(unique = true, nullable = false)
+    private String name;
+
+    @Column(unique = true, nullable = false)
+    private String color;
+
+    @ManyToOne
+    @JoinColumn(name = "up_station_id")
+    private Station upStation;
+
+    @ManyToOne
+    @JoinColumn(name = "down_station_id")
+    private Station downStation;
+
+    @Column(nullable = false)
+    private int distance;
+
+    protected Line() {
+    }
+
+    private Line(String name, String color, Station upStation, Station downStation, Integer distance) {
+        this.name = name;
+        this.color = color;
+        this.upStation = upStation;
+        this.downStation = downStation;
+        this.distance = distance;
+    }
+
+    public static Line of(String name, String color, Station upStation, Station downStation, Integer distance) {
+        return new Line(name, color, upStation, downStation, distance);
+    }
+
+    public void update(LineRequest lineRequest) {
```

Comment:
도메인이 Request를 바로 알기보다는 값만 받아오면 어떨까요...?
```suggestion
    public void update(String name, String color) {
```
Request 같은 dto의 변경이 도메인의 변경에 영향을 주는 구조는 피하는 경향이 있는 것 같습니다!

Reply:
아하 이 부분까진 미처 생각하지 못했는데, DTO 변경에 따라 `update`가 영향을 받을 수 있겠네요!